includes/admin/admin-pages.php

```diff
@@ -111,252 +111,166 @@ function give_add_options_links() {
-					if ( ( 'give_forms' == $typenow || 'give_forms' === $post_type ) || 'give_forms' === $post_type || ( 'post-new.php' == $pagenow && 'give_forms' === $post_type ) ) {
-						$found = true;
-					}
-					break;
-			}
-			break;
-		case 'categories':
-			switch ( $passed_view ) {
-				case 'list-table':
-				case 'new':
-					if ( ( 'give_forms' == $typenow || 'give_forms' === $post_type ) && $pagenow == 'edit-tags.php' && 'edit' !== $action && 'give_forms_category' === $taxonomy ) {
-						$found = true;
-					}
-					break;
-				case 'edit':
-					if ( ( 'give_forms' == $typenow || 'give_forms' === $post_type ) && $pagenow == 'edit-tags.php' && 'edit' === $action && 'give_forms_category' === $taxonomy ) {
-						$found = true;
-					}
-					break;
-				default:
-					if ( ( 'give_forms' == $typenow || 'give_forms' === $post_type ) && $pagenow == 'edit-tags.php' && 'give_forms_category' === $taxonomy ) {
-						$found = true;
-					}
-					break;
-			}
-			break;
-		case 'tags':
-			switch ( $passed_view ) {
-				case 'list-table':
-				case 'new':
-					if ( ( 'give_forms' == $typenow || 'give_forms' === $post_type ) && $pagenow == 'edit-tags.php' && 'edit' !== $action && 'give_forms_tag' === $taxonomy ) {
-						$found = true;
-					}
-					break;
-				case 'edit':
-					if ( ( 'give_forms' == $typenow || 'give_forms' === $post_type ) && $pagenow == 'edit-tags.php' && 'edit' === $action && 'give_forms_tag' === $taxonomy ) {
-						$found = true;
-					}
-					break;
-				default:
-					if ( ( 'give_forms' == $typenow || 'give_forms' === $post_type ) && $pagenow == 'edit-tags.php' && 'give_forms_tag' === $taxonomy ) {
-						$found = true;
-					}
-					break;
-			}
-			break;
-		case 'payments':
-			switch ( $passed_view ) {
-				case 'list-table':
-					if ( ( 'give_forms' == $typenow || 'give_forms' === $post_type ) && $pagenow == 'edit.php' && 'give-payment-history' === $page && false === $view ) {
-						$found = true;
-					}
-					break;
-				case 'edit':
-					if ( ( 'give_forms' == $typenow || 'give_forms' === $post_type ) && $pagenow == 'edit.php' && 'give-payment-history' === $page && 'view-payment-details' === $view ) {
-						$found = true;
-					}
-					break;
-				default:
-					if ( ( 'give_forms' == $typenow || 'give_forms' === $post_type ) && $pagenow == 'edit.php' && 'give-payment-history' === $page ) {
+	$found          = false;
+	$get_query_args = isset( $_GET ) ? array_map( 'give_clean', $_GET ) : array();
+
+	// Set default argument, if not passed.
+	$query_args = wp_parse_args( $get_query_args, array_fill_keys( array( 'post_type', 'action', 'taxonomy', 'page', 'view', 'tab' ), false ) );
+
+	// Page expected to pass.
+	$expected_pages = array( 'give_forms', 'categories', 'tags', 'payments', 'reports', 'settings', 'addons', 'donors' );
+
+	// If post type or type is give_forms.
+	if (
+		( 'give_forms' === $typenow || 'give_forms' === $query_args['post_type'] )
+		|| in_array( $passed_page, $expected_pages, true )
+	) {
```

Comment:
Rather than nesting everything in this can you do the opposite condition and pass `$found = false`?
I would prefer less nesting.

Reply:
That will be great, I am working on this. @DevinWalker

Reply:
What's the status on this @emgk ?